src/thread.ts

```diff
@@ -354,17 +354,15 @@ export class Thread<SCG extends ExtendableGenerics = DefaultGenerics> {
       sortedArray: replies,
       sortDirection: 'ascending',
       selectValueToCompare: (reply) => reply.created_at.getTime(),
+      selectKey: (reply) => reply.id,
     });
 
-    const actualIndex =
-      replies[index]?.id === message.id ? index : replies[index - 1]?.id === message.id ? index - 1 : null;
-
-    if (actualIndex === null) {
+    if (replies[index]?.id !== message.id ? index : null) {
       return;
     }
```

Comment:
I think we can ditch this whole thing, no?

Reply:
I don't think so, the index returned by `findIndexInSortedArray` is not necessarily pointing at an exact match.

Reply:
The confusing part is the rest of the condition (`? index : null`) - in this case it'd go to the early return if the message id's match and the index is non-zero. 🤔

Reply:
Oh man, for real. Missed that :)

Reply:
Fixed in https://github.com/GetStream/stream-chat-js/pull/1421/commits/17f8269c543266c59939915df0fa8849995bb122.